src/Components/Aspire.Azure.Messaging.ServiceBus/AspireServiceBusExtensions.cs

```diff
@@ -317,17 +305,17 @@ protected override IAzureClientBuilder<ServiceBusReceiver, ServiceBusReceiverOpt
                     throw new InvalidOperationException($"A ServiceBusReceiver could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a 'ConnectionString' or 'Namespace' in the '{configurationSectionName}' configuration section.");
                 }
 
-                if (string.IsNullOrEmpty(settings.QueueName) && (string.IsNullOrEmpty(settings.TopicName) || string.IsNullOrEmpty(settings.SubscriptionName)))
+                if (string.IsNullOrEmpty(settings.QueueOrTopicName) && string.IsNullOrEmpty(settings.SubscriptionName))
                 {
-                    throw new InvalidOperationException($"A ServiceBusReceiver could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a 'QueueName' or 'TopicName' and 'SubscriptionName' in the '{configurationSectionName}' configuration section.");
+                    throw new InvalidOperationException($"A ServiceBusReceiver could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a 'QueueOrTopicName' or 'SubscriptionName' in the '{configurationSectionName}' configuration section.");
```

Comment:
For subscriptions, don't you need both the QueueOrTopicName and the SubscriptionName?

Reply:
Yeah, this error message is kinda hard to word now that we've combined the queue and topic name.

If you provide just the `QueueOrTopicName`, then it creates a receiver assuming that the value represents a queue.

If you provide a subscription ID, it assumes that it's a topic + subscription pairing.

Maybe the right phrasing here is `QueueOrTopicName` and/or `SubscriptionName`.

Reply:
`specify a 'QueueOrTopicName' and if using a subscription, the 'SubscriptionName' ...`?